src/synapse/api.py

```diff
@@ -73,16 +92,21 @@ def _generate_mac(
     nonce: str,
     user: str,
     password: str,
-    admin: bool = False,
+    admin: str | bool = False,
     user_type: typing.Optional[str] = None,
 ) -> str:
     """Generate mac to register user.
 
+    "The MAC is the hex digest output of the HMAC-SHA1 algorithm, with the key being the shared
+    secret and the content being the nonce, user, password, either the string "admin" or "notadmin"
+    , and optionally the user_type each separated by NULs.".
```

Comment:
comma on previous line?

Reply:
If I change will raise line too long.

Reply:
what if `"notadmin"` is moved to the next line then?